certification/internal/shell/base_on_ubi.go

```diff
@@ -1,23 +1,36 @@
 package shell
 
 import (
-	"os/exec"
 	"strings"
 
 	"github.com/komish/preflight/certification"
+	"github.com/komish/preflight/cli"
 	"github.com/sirupsen/logrus"
 )
 
 type BaseOnUBICheck struct{}
 
 func (p *BaseOnUBICheck) Validate(image string, logger *logrus.Logger) (bool, error) {
-	stdouterr, err := exec.Command("podman", "run", "--rm", "-it", "--entrypoint", "cat", image, "/etc/os-release").CombinedOutput()
+	podmanEngine := PodmanCLIEngine{}
+	return p.validate(podmanEngine, image, logger)
+}
+
+func (p *BaseOnUBICheck) validate(podmanEngine cli.PodmanEngine, image string, logger *logrus.Logger) (bool, error) {
```

Comment:
I was wondering if we can declare the podmanEngine inside the old Validate() method and have one function instead of two?

Reply:
This is purposeful. This allows us to change out the PodmanEngine with another class that implements the interface, like FakePodmanEngine does. It would be harder to do so if we left the actual implementation in Validate.